src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java

```diff
@@ -158,9 +159,9 @@ public ResponseEntity getStockCards(@PathVariable Long facilityId,
     @RequestMapping(value = "facilities/{facilityId}/stockCards", method = POST, headers = ACCEPT_JSON)
     @ApiOperation(value="Update stock cards at a facility.")
     @Transactional
-    public ResponseEntity adjustStock(@PathVariable long facilityId,
-                                      @RequestBody(required=true) List<StockEvent> events,
-                                      HttpServletRequest request) {
+    public ResponseEntity processStock(@PathVariable long facilityId,
```

Comment:
The processStock method relies on a private method called getLotOnHand, and defined in this controller. I wonder, though, whether the getLotOnHand method better belongs in the service-layer. It has a fair amount of logic in it – perhaps more than belongs at the controller level.

Also, the getLotOnHand method makes interesting use of a StringBuilder to return errors to the user. Is this done elsewhere in the System? At first glance, a developer might think a StringBuilder is used rather than Java’s inbuilt try-catch mechanism because we want the ability to build and return a list of multiple errors. That’s not the case in this instance, however, so I wonder if the standard try-catch approach may be preferable.